src/main/scala/io/sdkman/changelogs/java/SapMachineMigrations.scala

```diff
@@ -65,57 +65,53 @@ class SapMachineMigrations {
     )
 
   @ChangeSet(
-    order = "0017",
-    id = "0017-add_sapmchn_jdk_11.0.9",
-    author = "poad"
+    order = "0021",
+    id = "0021-add_sapmchn_jdk_15.0.1",
+    author = "eddumelendez"
   )
-  def migrate0017(implicit db: MongoDatabase) =
+  def migrate0021(implicit db: MongoDatabase) =
     Map(
       Linux64 -> "linux-x64_bin.tar.gz",
+      MacOSX  -> "osx-x64_bin.tar.gz",
       Windows -> "windows-x64_bin.zip"
     ).map {
-        case (platform, suffix) =>
-          toVersion(platform, suffix, "11.0.9")
+        case (platform, binary) =>
+          toVersion(platform, binary, "15.0.1")
       }
       .toList
       .validate()
       .insert()
-      .foreach { version =>
-        removeVersion("java", "11.0.8-sapmchn", version.platform)
-      }
 
   @ChangeSet(
-    order = "0019",
-    id = "0019-add_sapmchn_jdk_11.0.9_mac",
+    order = "0022",
+    id = "0022-remove_sapmchn_jdk_11.0.9.0",
     author = "poad"
   )
-  def migrate0019(implicit db: MongoDatabase) =
-    Map(
-      MacOSX -> "osx-x64_bin.tar.gz"
-    ).map {
-        case (platform, suffix) =>
-          toVersion(platform, suffix, "11.0.9")
-      }
-      .toList
-      .validate()
-      .insert()
-      .foreach { version =>
-        removeVersion("java", "11.0.8-sapmchn", version.platform)
+  def migrate0022(implicit db: MongoDatabase) =
+    Set(Linux64, Windows, MacOSX)
+      .foreach { platform =>
+        removeVersion("java", "11.0.9-sapmchn", platform)
       }
 
   @ChangeSet(
-    order = "0021",
-    id = "0021-add_sapmchn_jdk_15.0.1",
-    author = "eddumelendez"
+    order = "0023",
+    id = "0023-add_sapmchn_jdk_11.0.9_1",
+    author = "poad"
   )
-  def migrate0021(implicit db: MongoDatabase) =
+  def migrate0023(implicit db: MongoDatabase) =
     Map(
       Linux64 -> "linux-x64_bin.tar.gz",
-      MacOSX  -> "osx-x64_bin.tar.gz",
-      Windows -> "windows-x64_bin.zip"
+      Windows -> "windows-x64_bin.zip",
+      MacOSX  -> "osx-x64_bin.tar.gz"
     ).map {
         case (platform, binary) =>
-          toVersion(platform, binary, "15.0.1")
+          Version(
+            "java",
+            "11.0.9.1-sapmchn",
```

Comment:
I think we should keep

```suggestion
            "11.0.9-sapmchn",
```